examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -242,6 +242,7 @@ CHIP_ERROR PairingCommand::PairWithCode(NodeId remoteId)
     }
 
     ReturnErrorOnFailure(MaybeDisplayTermsAndConditions(commissioningParams));
+    CurrentCommissioner().RegisterPairingDelegate(this);
```

Comment:
Why?  If we are here, we got called from RunInternal, which got called from `PairingCommand::RunCommand` which already did this.

Reply:
I found the mPairingDelegate won't be set without this that PairingCommand::OnPairingComplete() won't be called

Reply:
How, exactly?  Again, our caller does this, right?

But again: nothing to do with signal handlers.